File: Unit2-06/Typescript/Airplane.ts
/*
 * This program is an airplane class.
 *
 * @author  Johanna Liu
 * @version 1.0
 * @since   2024-04-24
 */

// This is the airplane class.
export class Airplane {
  /** The internal variable that stores speed. */
  private speed: number

  // Constructor.
  public Airplane() {
  }

  // Set an airplane speed.
  public setSpeed(speed: number) {
      this.speed = speed
  }

  // Get speed.
  public getSpeed(): number {
      return this.speed
  }
}

File: Unit2-06/Java/Airplane.java
/*
* This program is the airplane class.
*
* @author  Johanna Liu
* @version 1.0
* @since   2024-05-03
*/

/**
 * This is the class.
*/
public class Airplane {
    /**
     * Uses the airplane's speed.
     *
     * @param speed of the airplane
    */
    private int speed;

    /**
     * Creates the base airplane.
    */
    public Airplane() {
    }

    /**
     * Checks the speed of the airplane.
     *
     * @return the speed of the airplane
    */
    public int getSpeed() {
        return this.speed;
    }

    /**
     * Sets the speed of the airplane.
     *
     * @param newSpeed of the airplane
    */
    public void setSpeed(int newSpeed) {
        this.speed = newSpeed;
    }
}